### PR TITLE
OCT-620 - stop non-problem publications being linkable to topics

### PR DESCRIPTION
**Pull request by finlay-jisc**
The purpose of this PR was to correct a bug with the original work on OCT-620. Only research problems should have the option to link to a topic, but in testing, any publication type could do this.

---

### Acceptance Criteria:

Only research problems can be linked to a topic using the UI.

---

### Checklist:

- [x] Local manual testing conducted
- [x] Automated tests added
- [ ] Documentation updated

### CODE REVIEW

File: ui/src/components/Publication/Creation/LinkedItems/index.tsx [hunk 1]
```diff
@@ -113,34 +113,36 @@ const Links: React.FC = (): React.ReactElement => {
...
-                            const value: Types.LinkedEntityType = e.target.value as Types.LinkedEntityType;
-                            setEntityType(value);
-                        }}
-                        value={entityType}
-                        className="block rounded-md border border-grey-200 outline-none focus:ring-2 focus:ring-yellow-500 sm:mr-0"
-                    >
-                        <option value="PUBLICATION">Publications</option>
-                        <option value="TOPIC">Research topics</option>
-                    </select>
+                    {!!isProblem && (
```
Comment (florin-holhos) [inline ui/src/components/Publication/Creation/LinkedItems/index.tsx:116]: "isProblem" is already a boolean. Does it need to be converted to a boolean again?

File: ui/src/components/Publication/Creation/LinkedItems/index.tsx [hunk 2]
```diff
@@ -156,7 +158,7 @@ const Links: React.FC = (): React.ReactElement => {
                     entityType="PUBLICATION"
                 />
             )}
-            {!error && !!topics.length && (
+            {!error && !!topics.length && !!isProblem && (
```
Comment (florin-holhos) [inline ui/src/components/Publication/Creation/LinkedItems/index.tsx:161]: Topics length will always be 0 for other publication types other than PROBLEM.
'&& !!isProblem' shouldn't make any difference.

**Review by florin-holhos — CHANGES_REQUESTED** [review]
On the API side, we need to make sure we don't allow people to add topics to other publication types beside PROBLEM. Whenever we create/update a publication, if we get topics into the request body, we need to check if the publication types is PROBLEM, otherwise the API should return an error. I managed to add topics to a Hypothesis using Postman.

File: ui/src/components/Publication/Creation/LinkedItems/index.tsx (hunk 1 above)
Comment (finlay-jisc) [reply ui/src/components/Publication/Creation/LinkedItems/index.tsx]: No, there is no need

**Pull request merged**